### PR TITLE
Implemented scanner on purchase order detail page(#1xx6hbj)

### PR DESCRIPTION
**Pull request by disha1202**
### Related Issues

Closes #

### Screenshots of Visual Changes before/after (If There Are Any)

**IMPORTANT NOTICE** - Remember to add changelog entry

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/hotwax/receiving#contribution-guideline)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
     "moment-timezone": "^0.5.33",
     "vue-barcode-reader": "^1.0.1",
     "vue": "^3.2.26",
+    "vue-barcode-reader": "^1.0.1",
```
Comment (adityasharma7) [inline package.json:29]: I think this is added twice. Check line 27

File: src/services/PurchaseOrderService.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import api from '@/api';
```
Comment (adityasharma7) [inline src/services/PurchaseOrderService.ts:1]: I think this is now OrderService. Check the latest code  
https://github.com/hotwax/receiving/pull/56/files#diff-42325f64b3fb9a1683655b01d10fe216c4adf02ba7d1838439e0da1d5c499f4dR11

https://github.com/hotwax/receiving/blob/main/src/services/OrderService.ts

File: src/store/modules/order/actions.ts [hunk 3]
```diff
@@ -33,6 +37,14 @@ const actions: ActionTree<OrderState, RootState> = {
     }
     return resp;
   },
+  async updateProductCount({ commit, state }, payload ) {
+    state.current.items.forEach((item: any) => {
```
Comment (adityasharma7) [inline src/store/modules/order/actions.ts:41]: I think we should use find and update the qunatity

File: src/store/modules/order/mutations.ts [hunk 4]
```diff
@@ -8,9 +8,14 @@ const mutations: MutationTree <OrderState> = {
     state.purchaseOrders.total = payload.total;
   },
   [types.ORDER_CURRENT_UPDATED](state, payload) {
-    state.current.orderId = payload.order[0]?.orderId;
-    state.current.externalOrderId = payload.order[0]?.externalOrderId;
-    state.current.items = payload.order
+    if(payload.order){
```
Comment (adityasharma7) [inline src/store/modules/order/mutations.ts:11]: Why this is changed?

File: src/store/modules/purchaseOrder/PurchaseOrderState.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+export default interface PurchaseOrderState {
+  purchaseOrders: {
```
Comment (adityasharma7) [inline src/store/modules/purchaseOrder/PurchaseOrderState.ts:2]: I think this needs to be reverted

File: src/store/modules/purchaseOrder/actions.ts [hunk 6]
```diff
@@ -0,0 +1,82 @@
+import { PurchaseOrdersService } from "@/services/PurchaseOrderService";
+import { ActionTree } from 'vuex'
```
Comment (adityasharma7) [inline src/store/modules/purchaseOrder/actions.ts:2]: I think this needs to be reverted and moved to order actions

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/store/modules/order/mutations.ts (hunk 4 above)
Comment (disha1202) [reply src/store/modules/order/mutations.ts:11]: Initially when we are setting the current order we are passing an object that contains an array order as payload but when we are updating the quantity accepted of an item we are passing state.current as payload.

File: src/store/modules/purchaseOrder/PurchaseOrderState.ts (hunk 5 above)
Comment (disha1202) [reply src/store/modules/purchaseOrder/PurchaseOrderState.ts]: Reverted.

File: src/store/modules/order/actions.ts (hunk 3 above)
Comment (disha1202) [reply src/store/modules/order/actions.ts]: Updated the code to use find and update the quantity.

File: package.json (hunk 1 above)
Comment (disha1202) [reply package.json]: Reverted this change.

File: src/services/PurchaseOrderService.ts (hunk 2 above)
Comment (disha1202) [reply src/services/PurchaseOrderService.ts]: Yes Sir, reverted these changes

File: src/store/modules/purchaseOrder/actions.ts (hunk 6 above)
Comment (disha1202) [reply src/store/modules/purchaseOrder/actions.ts]: Reverted.

**Review by ymaheshwari1 — APPROVED** [review]

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**